src/EnergyPlus/UnitarySystem.cc

```diff
@@ -1161,6 +1161,137 @@ namespace UnitarySystems {
+                SysSensCoolingLoad = state.dataSize->FinalZoneSizing(state.dataSize->CurZoneEqNum).DesCoolLoad;
+                SysLatCoolingLoad = state.dataSize->CalcFinalZoneSizing(state.dataSize->CurZoneEqNum).DesLatentCoolLoad;
+                SysTotCoolingLoad = SysSensCoolingLoad + SysLatCoolingLoad;
+                SysHeatingLoad = state.dataSize->FinalZoneSizing(state.dataSize->CurZoneEqNum).DesHeatLoad;
+            }
+
+            if (this->m_HeatPump &&
+                ((state.dataSize->CurSysNum > 0 && !state.dataSize->FinalSysSizing.empty() &&
+                  state.dataSize->FinalSysSizing(state.dataSize->CurSysNum).heatCoilSizingMethod != DataSizing::HeatCoilSizMethod::None) ||
+                 (state.dataSize->CurZoneEqNum > 0 && !state.dataSize->FinalZoneSizing.empty() &&
+                  state.dataSize->FinalZoneSizing(state.dataSize->CurZoneEqNum).heatCoilSizingMethod != DataSizing::HeatCoilSizMethod::None))) {
+                std::string_view cCoilName;
+                std::string_view hCoilName;
+                Real64 coilSHR = 1.0;
+                if (this->m_CoolingCoilType_Num == HVAC::Coil_CoolingWaterToAirHPSimple) {
+                    auto const &thisCoil = state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(this->m_CoolingCoilIndex);
+                    cCoilName = thisCoil.Name;
+                    coilSHR = 0.8;
+                } else if (this->m_CoolingCoilType_Num == HVAC::Coil_CoolingWaterToAirHPVSEquationFit ||
+                           this->m_CoolingCoilType_Num == HVAC::Coil_CoolingAirToAirVariableSpeed) {
+                    auto const &thisCoil = state.dataVariableSpeedCoils->VarSpeedCoil(this->m_CoolingCoilIndex);
+                    cCoilName = thisCoil.Name;
+                    coilSHR = thisCoil.MSRatedSHR(thisCoil.NumOfSpeeds);
+                } else if (this->m_CoolingCoilType_Num == HVAC::CoilDX_Cooling) {
+                    auto const &thisCoil = state.dataCoilCoolingDX->coilCoolingDXs[this->m_CoolingCoilIndex];
+                    cCoilName = thisCoil.name;
+                    coilSHR = thisCoil.performance->grossRatedSHR(state);
+                } else {
+                    if (!state.dataDXCoils->DXCoil.empty()) {
+                        auto const &thisCoil = state.dataDXCoils->DXCoil(this->m_CoolingCoilIndex);
+                        cCoilName = thisCoil.Name;
+                        coilSHR = (thisCoil.NumOfSpeeds == 0) ? thisCoil.RatedSHR(1) : thisCoil.RatedSHR(thisCoil.NumOfSpeeds);
+                    } else {
+                        ShowSevereError(state,
+                                        format("Developer Error in Heat Pump ACCA Sizing: cooling coil not found for {}:{} with coil type = {}.",
+                                               this->UnitType,
+                                               this->Name,
+                                               HVAC::cAllCoilTypes(this->m_CoolingCoilType_Num)));
+                    }
```

Comment:
Added a safety (if not empty) for this last else, which includes multiple DXCoil types, so that this would not crash (as it did many time during development) and instead give an indication of what type of coil was missed so that coil type can be added.